### PR TITLE
healtcheck setup and apis

### PR DESCRIPTION
**Pull request by Dinaamagdy**

### CODE REVIEW

File: tests/0_orchestrator/test_suite/framework/orchestrator_apis/health_apis.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    @catch_exception_decoration
+    def get_all_nodes_health(self):
+        return self.orchestrator_client.health.ListNodesHealth()
+
+    @catch_exception_decoration
+    def get_node_health(self, node_id):
+        return self.orchestrator_client.health.ListNodeHealth(nodeid=node_id)
+
+    @catch_exception_decoration
+    def get_storageclusters_health(self, node_id):
```
Comment (islamTaha12) [inline tests/0_orchestrator/test_suite/framework/orchestrator_apis/health_apis.py]: useless node_id?!

File: tests/0_orchestrator/test_suite/testcases/basic_tests/test12_health_apis.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+class TesthealthcheckAPI(TestcasesBase):
+    @classmethod
+    def setUpClass(cls):
+        self = cls()
+        super(TesthealthcheckAPI, self).setUp()
+
+        self.lg.info("Get list of nodes . ")
+        response = self.nodes_api.get_nodes()
+        self.assertEqual(response.status_code, 200)
+        Nodes_result = response.json()
```
Comment (islamTaha12) [inline tests/0_orchestrator/test_suite/testcases/basic_tests/test12_health_apis.py]: small letters

File: tests/0_orchestrator/test_suite/testcases/basic_tests/test12_health_apis.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+        self.lg.info("Get list of a nodes healthcheck api request.")
+        response = self.healthcheck_api.get_all_nodes_health()
+        self.assertEqual(response.status_code, 200)
+        health_result = response.json()
+
+        self.lg.info(" Check that all nodes have health check, should succeed. ")
```
Comment (islamTaha12) [inline tests/0_orchestrator/test_suite/testcases/basic_tests/test12_health_apis.py]: even if there is halted nodes?!

File: tests/0_orchestrator/test_suite/testcases/basic_tests/test12_health_apis.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import random
+import time
+import unittest
+from testcases.testcases_base import TestcasesBase
+
+
+class TesthealthcheckAPI(TestcasesBase):
+    @classmethod
+    def setUpClass(cls):
```
Comment (islamTaha12) [inline tests/0_orchestrator/test_suite/testcases/basic_tests/test12_health_apis.py]: what is the purpose of this class method?

File: tests/0_orchestrator/test_suite/framework/orchestrator_apis/health_apis.py (hunk 1 above)
Comment (Dinaamagdy) [reply tests/0_orchestrator/test_suite/framework/orchestrator_apis/health_apis.py]: done

File: tests/0_orchestrator/test_suite/testcases/basic_tests/test12_health_apis.py (hunk 4 above)
Comment (Dinaamagdy) [reply tests/0_orchestrator/test_suite/testcases/basic_tests/test12_health_apis.py]: to check that all nodes have healtcheck

File: tests/0_orchestrator/test_suite/testcases/basic_tests/test12_health_apis.py (hunk 2 above)
Comment (Dinaamagdy) [reply tests/0_orchestrator/test_suite/testcases/basic_tests/test12_health_apis.py]: done

File: tests/0_orchestrator/test_suite/testcases/basic_tests/test12_health_apis.py (hunk 3 above)
Comment (Dinaamagdy) [reply tests/0_orchestrator/test_suite/testcases/basic_tests/test12_health_apis.py]: yes , halted nodes should have heclthchek and it's status halted

File: tests/0_orchestrator/test_suite/testcases/basic_tests/test12_health_apis.py (hunk 4 above)
Comment (islamTaha12) [reply tests/0_orchestrator/test_suite/testcases/basic_tests/test12_health_apis.py]: It should be a test case not a setup

**Pull request merged**